pkg/resource_builders/envoyconfig/listeners.go

```diff
@@ -0,0 +1,260 @@
+		ListenerFilters: ListenerFilters_v1(opts.CertificateSecretName != nil),
+		FilterChains: []*envoy_config_listener_v3.FilterChain{{
+			Filters: []*envoy_config_listener_v3.Filter{{
+				Name: "envoy.filters.network.http_connection_manager",
+				ConfigType: &envoy_config_listener_v3.Filter_TypedConfig{
+					TypedConfig: func() *anypb.Any {
+						any, err := anypb.New(
+							&http_connection_manager_v3.HttpConnectionManager{
+								AccessLog: AccessLogConfig_v1(desc.GetName(), opts.CertificateSecretName != nil),
+								CommonHttpProtocolOptions: func() *envoy_config_core_v3.HttpProtocolOptions {
+									po := &envoy_config_core_v3.HttpProtocolOptions{
+										IdleTimeout: durationpb.New(3600 * time.Second),
+									}
+									if opts.MaxConnectionDuration != nil {
+										po.MaxConnectionDuration = durationpb.New(opts.MaxConnectionDuration.Duration)
+									}
+									if opts.AllowHeadersWithUnderscores != nil && *opts.AllowHeadersWithUnderscores {
+										po.HeadersWithUnderscoresAction = envoy_config_core_v3.HttpProtocolOptions_ALLOW
+									} else {
+										po.HeadersWithUnderscoresAction = envoy_config_core_v3.HttpProtocolOptions_REJECT_REQUEST
+									}
+									return po
+								}(),
+
+								HttpFilters: HttpFilters_v1(opts.RateLimitOptions),
+								HttpProtocolOptions: func() *envoy_config_core_v3.Http1ProtocolOptions {
+									if opts.DefaultHostForHttp10 != nil {
+										return &envoy_config_core_v3.Http1ProtocolOptions{
+											AcceptHttp_10:         true,
+											DefaultHostForHttp_10: *opts.DefaultHostForHttp10,
+										}
+									}
+									return &envoy_config_core_v3.Http1ProtocolOptions{}
+								}(),
+								Http2ProtocolOptions: &envoy_config_core_v3.Http2ProtocolOptions{
+									MaxConcurrentStreams:        wrapperspb.UInt32(100),
+									InitialStreamWindowSize:     wrapperspb.UInt32(65536),   // 64 KiB
+									InitialConnectionWindowSize: wrapperspb.UInt32(1048576), // 1 MiB
+								},
+								RequestTimeout:    durationpb.New(300 * time.Second),
+								RouteSpecifier:    RouteConfigFromAds_v1(opts.RouteConfigName),
+								StatPrefix:        desc.GetName(),
+								StreamIdleTimeout: durationpb.New(300 * time.Second),
+								UseRemoteAddress:  wrapperspb.Bool(true),
+							})
+						if err != nil {
+							panic(err)
+						}
+						return any
+					}(),
+				},
+			}},
+		}},
+		PerConnectionBufferLimitBytes: wrapperspb.UInt32(32768), // 32 KiB
+	}
+
+	// Apply TLS config if this is a HTTPS listener
+	if opts.CertificateSecretName != nil {
+		listener.FilterChains[0].TransportSocket = TransportSocket_v1(*opts.CertificateSecretName, *opts.EnableHttp2)
+	}
+
+	return listener, nil
+}
+
+func ListenerFilters_v1(tls bool) []*envoy_config_listener_v3.ListenerFilter {
+	filters := []*envoy_config_listener_v3.ListenerFilter{}
+	if tls {
+		filters = append(filters, &envoy_config_listener_v3.ListenerFilter{
+			Name: "envoy.filters.listener.tls_inspector",
```

Comment:
Is it not needed:
```
              typed_config:
                "@type": type.googleapis.com/envoy.extensions.filters.listener.tls_inspector.v3.TlsInspector
```
?

Reply:
We could actually drop this filter because we are not using it. It is only useful for tls connections, to match a request to a filter chain using SNI. I just left it in case we do something like that in the future.

Reply:
nice, fair enough!